### PR TITLE
LibSQL: Rewrite the SQL::Value type to be contained within one class

### PR DESCRIPTION
**Pull request by trflynn89**
Reviving some old branches to make LibSQL more feasible to be used to e.g. store cookies from the Browser.

Currently, the Value class is essentially a "pImpl" wrapper around the
ValueImpl hierarchy of classes. This is a bit difficult to follow and
reason about, as methods jump between the Value class and its impl
classes.

This changes the Variant held by Value to instead store the specified
types (String, int, etc.) directly. In doing so, the ValueImpl classes
are removed, and all methods are now just concise Variant visitors.

As part of this rewrite, support for the "array" type is dropped (or
rather, just not re-implemented) as it was unused. If it's needed in the
future, support can be re-added.

This does retain the ability for non-NULL types to store NULL values
(i.e. an empty Optional). I tried dropping this support as well, but it
is depended upon by the on-disk storage classes in non-trivial ways.

### CODE REVIEW

File: Userland/Libraries/LibSQL/AST/Expression.cpp [hunk 1]
```diff
@@ -46,7 +42,7 @@ ResultOr<Value> ChainedExpression::evaluate(ExecutionContext& context) const
     for (auto& expression : expressions())
         values.unchecked_append(TRY(expression.evaluate(context)));
 
-    return Value { move(values) };
+    return TRY(Value::create_tuple(move(values)));
```
Comment (alimpfard) [inline Userland/Libraries/LibSQL/AST/Expression.cpp:45]: No need for TRY,
```suggestion
    return Value::create_tuple(move(values));
```
should also work.

File: Userland/Libraries/LibSQL/Value.cpp [hunk 2]
```diff
@@ -1,301 +1,340 @@
...
 
 Optional<u32> Value::to_u32() const
 {
-    if (is_null())
-        return {};
-    auto ret = to_int();
-    if (ret.has_value())
-        return static_cast<u32>(ret.value());
+    if (auto result = to_int(); result.has_value())
+        return static_cast<u32>(result.value());
```
Comment (alimpfard) [inline Userland/Libraries/LibSQL/Value.cpp:151]: How about negative values?

Comment (alimpfard) [reply Userland/Libraries/LibSQL/Value.cpp:151]: I guess it's an existing bug, so maybe just leave a note?

**Review by alimpfard — COMMENTED** [review]
Just a nitpick or two

File: Userland/Libraries/LibSQL/Value.cpp (hunk 2 above)
Comment (trflynn89) [reply Userland/Libraries/LibSQL/Value.cpp:151]: Yeah, there's a few suspect signed/unsigned things in this file I marked with FIXMEs. Slapped another one here.

**Pull request merged**